ndb/src/google/cloud/ndb/_ports.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Ported implementations from the Google App Engine SDK.
+
+The following members have been brought in:
+
+* ``google.appengine.api.datastore_types.BlobKey``
+"""
+
+import functools
+
+from google.cloud.ndb import exceptions
+
+
+_MAX_STRING_LENGTH = 1500
+
+
+@functools.total_ordering
+class BlobKey:
+    """Key used to identify a blob in the blobstore.
+
+    .. note::
+
+        The blobstore was an early Google App Engine feature that later became
+        Google Cloud Storage.
```

Comment:
Can this also be used with Cloud Storage?

Reply:
Blobstore and Google Cloud Storage (GCS) are the same thing. But blobstore came first, and is a GAE feature that wraps around GCS. (It originally was just a blob store feature, i.e. no wrapping.)

Reply:
I realize that didn't really answer your question.

The answer is essentially "not really". I'd imagine we'll have to do a lot of work to get this to work the way we need it to, but we'll find out.

However, this property itself just stores some bytes in a column in the datastore.

Reply:
The reason I ask is because I'm wondering if this is worth doing. I could see dropping the integration with blobstore for now and adding integration with GCS as a new feature. If this code isn't useful for a future GCS integration, then I'm wondering why we need it at all.